lib/spack/spack/platforms/darwin.py

```diff
@@ -4,19 +4,24 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 import platform
+import llnl.util.cpu as cpu
 from spack.architecture import Platform, Target
 from spack.operating_systems.mac_os import MacOs
 
 
 class Darwin(Platform):
     priority    = 89
-    front_end   = 'x86_64'
-    back_end    = 'x86_64'
-    default     = 'x86_64'
 
     def __init__(self):
         super(Darwin, self).__init__('darwin')
-        self.add_target(self.default, Target(self.default))
+
+        for name in cpu.supported_target_names():
+            self.add_target(name, Target(name))
+
+        self.default = cpu.detect_host().name
+        self.front_end = self.default
+        self.back_end = self.default
```

Comment:
Can we push this up even further, e.g. to `Platform`?

Reply:
Cray and BGQ treats this part differently - so we would need to tweak also those classes. Overall I'd delay this refactor to a different PR.